export/client/clients/redis-client.go

```diff
@@ -0,0 +1,362 @@
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	err := deleteRegistration(conn, reg.ID.Hex())
+	if err != nil {
+		return err
+	}
+
+	_, err = addRegistration(conn, &reg)
+	return err
+}
+
+// Get a registration by ID
+// UnexpectedError - problem getting in database
+// NotFound - no registration with the ID was found
+func (c *Client) RegistrationById(id string) (r export.Registration, err error) {
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	err = getObjectById(conn, id, unmarshalObject, &r)
+	return r, err
+}
+
+// Get a registration by name
+// UnexpectedError - problem getting in database
+// NotFound - no registration with the name was found
+func (c *Client) RegistrationByName(name string) (r export.Registration, err error) {
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	err = getObjectByHash(conn, EXPORT_COLLECTION+":name", name, unmarshalObject, &r)
+	return r, err
+}
+
+// Delete a registration by ID
+// UnexpectedError - problem getting in database
+// NotFound - no registration with the ID was found
+func (c *Client) DeleteRegistrationById(id string) error {
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	return deleteRegistration(conn, id)
+}
+
+// Delete a registration by name
+// UnexpectedError - problem getting in database
+// NotFound - no registration with the name was found
+func (c *Client) DeleteRegistrationByName(name string) error {
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	id, err := redis.String(conn.Do("HGET", EXPORT_COLLECTION+":name", name))
+	if err == redis.ErrNil {
+		return ErrNotFound
+	} else if err != nil {
+		return err
+	}
+
+	return deleteRegistration(conn, id)
+}
+
+func (c *Client) ScrubAllExports() (err error) {
+	conn := c.Pool.Get()
+	defer conn.Close()
+
+	return unlinkCollection(conn, EXPORT_COLLECTION)
+}
+
+func addRegistration(conn redis.Conn, r *export.Registration) (id bson.ObjectId, err error) {
+	if !r.ID.Valid() {
+		r.ID = bson.NewObjectId()
+	}
+	id = r.ID
+	rid := r.ID.Hex()
+
+	ts := time.Now().UnixNano() / int64(time.Millisecond) //MakeTimestamp()
+	if r.Created == 0 {
+		r.Created = ts
+	}
+	r.Modified = ts
+
+	m, err := marshalObject(r)
+	if err != nil {
+		return id, err
+	}
+
+	conn.Send("MULTI")
+	conn.Send("SET", rid, m)
+	conn.Send("ZADD", EXPORT_COLLECTION, 0, rid)
+	conn.Send("HSET", EXPORT_COLLECTION+":name", r.Name, rid)
+	_, err = conn.Do("EXEC")
+	return id, err
+}
+
+func deleteRegistration(conn redis.Conn, id string) error {
+	object, err := redis.Bytes(conn.Do("GET", id))
+	if err == redis.ErrNil {
+		return ErrNotFound
+	} else if err != nil {
+		return err
+	}
+
+	r := export.Registration{}
+	err = unmarshalObject(object, &r)
+
+	conn.Send("MULTI")
+	conn.Send("DEL", id)
+	conn.Send("ZREM", EXPORT_COLLECTION, id)
+	conn.Send("HDEL", EXPORT_COLLECTION+":name", r.Name)
+	_, err = conn.Do("EXEC")
+	return err
+}
+
+// ------------------------------------ "imports" ------------------------------
+const (
```

Comment:
Should be at the top of the file IMHO. Also, if you consider that this lowers the redability, please feel free to move strings directly into the code as it was before.